cvmfs/cache_extern.cc

```diff
@@ -538,7 +538,7 @@ void *ExternalCacheManager::MainRead(void *data) {
       continue;
     } else {
       PANIC(kLogSyslogErr | kLogDebug, "unexpected message %s",
-            msg->GetTypeName().c_str());
+            msg->GetTypeName().data());
```

Comment:
Here and down: are we sure these are NULL terminated?

Reply:
`std::string_view` doesn't have [`c_str()`](https://en.cppreference.com/w/cpp/string/basic_string/c_str), so either one has to make a copy, or use something like [`data()`](https://en.cppreference.com/w/cpp/string/basic_string_view/data). Your question is answered at https://protobuf.dev/support/migration/#string_view-return-type. So, it *can* be non-null terminated, but where it's just the result of `GetTypeName()`, it should always be null-terminated. The new implementation is [here]( https://github.com/protocolbuffers/protobuf/blob/88e11bd819938bfc3f8a4b09495774b8d2d50cd3/src/google/protobuf/message_lite.cc#L106-L118).

Reply:
That does seem fine, but so does making a copy - I'd suggest to copy it for the time being.

Reply:
Ok, I tried the simplest patch. Since it's just a couple of lines that need to be changed, should I leave that to you to choose how to do the copy?

Reply:
Yes, I can take it from here. Thanks for the report and patch though!

Reply:
Alright, I will close this then. Cheers,